src/components/TodosFiltered/TodosFiltered.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
+
+import { Todo } from '../../types/Todo';
+import { client } from '../../utils/fetchClient';
+
+type Props = {
+  todos: Todo[],
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>
+  setAllTodos: React.Dispatch<React.SetStateAction<Todo[]>>
+  setErrorMessage: (a: string) => void;
+  allTodos: Todo[];
+};
+
+export const TodosFilter: React.FC<Props> = ({
+  todos,
+  setTodos,
+  setAllTodos,
+  allTodos,
+  setErrorMessage,
+}) => {
+  enum Status {
+    ALL = 'all',
+    ACTIVE = 'active',
+    COMPLETED = 'completed',
+  }
+
+  const [filterBy, setFilterBy] = useState<Status>(Status.ALL);
+
+  const getFilteredTodos = (filter: Status) => {
+    let filteredTodos: Todo[] = [];
+
+    switch (filter) {
+      case Status.ALL:
+        filteredTodos = allTodos;
+        break;
+      case Status.ACTIVE:
+        filteredTodos = allTodos.filter(todo => !todo.completed);
+        break;
+      case Status.COMPLETED:
+        filteredTodos = allTodos.filter(todo => todo.completed);
+        break;
+      default:
+        break;
+    }
+
+    setTodos(filteredTodos);
+  };
+
+  const handleRemoveAllCompleted = async () => {
+    try {
+      const completedTodos = allTodos.filter(todo => todo.completed);
+
+      await Promise.all(completedTodos.map(todo => client.delete(`/todos/${todo.id}`)));
+
+      setAllTodos(prevTodos => prevTodos.filter(
+        todo => !completedTodos.includes(todo),
+      ));
+    } catch (error) {
+      setErrorMessage('Unable to remove completed todos');
+      // eslint-disable-next-line no-console
+      console.error('An error occurred:', error);
+    }
+  };
+
+  useEffect(() => {
+    setTodos(allTodos);
+  }, [allTodos]);
+
+  const handleSetFilteredTodos
+    = (filter: Status) => {
+      setFilterBy(filter);
+      getFilteredTodos(filter);
+    };
+
+  const completedTodoLength
```

Comment:
The completedTodoLength and uncompletedTodos are calculated every time the component re-renders. This is unnecessary computation if todos does not change. Consider using useMemo hook for these computations.